pkg/controller/elasticsearch/pdb/reconcile.go

```diff
@@ -19,17 +20,44 @@ import (
 	commonv1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/common/v1"
 	esv1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/elasticsearch/v1"
 	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/hash"
+	lic "github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/license"
 	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/metadata"
 	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/reconciler"
 	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/elasticsearch/label"
 	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/elasticsearch/sset"
 	"github.com/elastic/cloud-on-k8s/v3/pkg/utils/k8s"
 )
 
-// Reconcile ensures that a PodDisruptionBudget exists for this cluster, inheriting the spec content.
-// The default PDB we setup dynamically adapts MinAvailable to the number of nodes in the cluster.
+// Reconcile ensures that PodDisruptionBudget(s) exists for this cluster, inheriting the spec content.
+//
+// For non-enterprise users: The default PDB we setup dynamically adapts MinAvailable to the number of nodes in the cluster.
+// For enterprise users: We optimize the PDBs that we setup to speed up Kubernetes cluster operations such as upgrades as much as safely possible
+//
+//	by grouping statefulSets by associated Elasticsearch node roles into the same PDB, and then dynamically maxUnavailable
+//	according to whatever cluster health is optimal for the set of roles.
+//
 // If the spec has disabled the default PDB, it will ensure none exist.
 func Reconcile(ctx context.Context, k8sClient k8s.Client, es esv1.Elasticsearch, statefulSets sset.StatefulSetList, meta metadata.Metadata) error {
+	licenseChecker := lic.NewLicenseChecker(k8sClient, es.Namespace)
```

Comment:
The second parameter must be the operator namespace (not the Elasticsearch one).

Reply:
I think we are still using the Elasticsearch namespace here instead of the controller namespace: https://github.com/elastic/cloud-on-k8s/pull/8780/files#diff-7166470f6fb1603354bea209027f74680de6e68f66ea9979952b6e8ebed8fdccR41

Reply:
I think this is not fixed (or I'm missing something)